community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementationHandle.java

```diff
@@ -102,6 +104,13 @@ public boolean isUnderlyingTransaction( KernelTransaction tx )
     }
 
     @Override
+    public Stream<ExecutingQuery> executingQueries()
+    {
+        KernelStatement statement = tx.tryAcquireStatement();
```

Comment:
That is not quite safe usage of underlying transaction.
Couple of this with current way:
1. statement can be already closed when we calling `executingQueries` in one of the tx handles in a stream - so that will throw exception which is not catch anywhere currently.
2. Underlying transaction can already be reused so we will return queries for some other not expected transaction